src/neptune/internal/utils/git.py

```diff
@@ -90,3 +107,95 @@ def to_git_info(git_ref: Union[GitRef, GitRefDisabled]) -> Optional[GitInfo]:
         )
     except:  # noqa: E722
         return None
+
+
+class DiffTracker:
+    def __init__(self, repo: git.Repo):
+        self._repo = repo
+        self._head = self._repo.head
+        self._upstream_commit_sha = None
+
+    @property
+    def repo_dirty(self) -> bool:
+        return self._repo.is_dirty()
+
+    @classmethod
+    def from_git_ref(cls, git_ref: Union[GitRef, GitRefDisabled]) -> Optional["DiffTracker"]:
+        if git_ref == GitRef.DISABLED:
+            return None
+
+        initial_repo_path = git_ref.resolve_path()
+        if initial_repo_path is None:
+            return None
+        try:
+            repo = get_git_repo(repo_path=initial_repo_path)
+        except (NoSuchPathError, InvalidGitRepositoryError):
+            repo = None
+
+        if repo:
```

Comment:
After the above suggestion -> no need to check it (`get_git_repo` returns always `Repo`)